core/src/main/java/com/netflix/iceberg/encryption/BaseEncryptionKeyMetadata.java

```diff
@@ -38,6 +38,6 @@ public ByteBuffer keyMetadata() {
   @Override
   public EncryptionKeyMetadata copy() {
     return new BaseEncryptionKeyMetadata(
-        ByteBuffers.copy(keyMetadata));
+        keyMetadata == null ? null : ByteBuffers.copy(keyMetadata));
```

Comment:
What about making `ByteBuffers.copy` null-safe? That would prevent future problems like this.